### PR TITLE
Improve scrolling / zooming behavior in flame charts

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Some of these changes will reduce the number of rebuilds, and one (increasing size of right padding for the last node in each row) is a band-aid fix for a larger and more difficult to solve issue (https://github.com/flutter/devtools/issues/2012).

### CODE REVIEW

File: packages/devtools_app/lib/src/charts/flame_chart.dart [hunk 1]
```diff
@@ -434,27 +432,28 @@ abstract class FlameChartState<T extends FlameChart,
...
+    // Calculate the new horizontal scroll position.
+    final newScrollOffset = fixedX >= 0
+        ? fixedX * currentZoom / previousZoom +
+            widget.startInset -
+            safeMouseHoverX
+        // We are in the fixed portion of the window - no need to transform.
+        : lastScrollOffset;
 
+    setState(() {
+      currentZoom = zoomController.value;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/charts/flame_chart.dart:456]: we already change currentZoom to this value outside of setState. I would avoid updating currentZoom until this line. Maybe add a 
nextZoom local variable to disambiguate.

File: packages/devtools_app/lib/src/charts/flame_chart.dart [hunk 2]
```diff
@@ -530,27 +529,24 @@ abstract class FlameChartState<T extends FlameChart,
...
+    // Call these in a post frame callback so that the scroll controllers have
+    // had time to update their scroll extents. Otherwise, we can hit a race
+    // where are trying to scroll to an offset that is beyond what the scroll
+    // controller thinks its max scroll extent is.
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      await Future.wait([
+        scrollHorizontallyToData(data),
+        if (scrollVertically) scrollVerticallyToData(data),
+      ]);
+    });
```
Comment (jacob314) [inline packages/devtools_app/lib/src/charts/flame_chart.dart:549]: why does zooToTimeRange impact the vertical scroll controller? I would think that would only impact horizontal.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/charts/flame_chart.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/charts/flame_chart.dart:549]: It doesn't impact the vertical scroll controller, but when we call scrollVerticallyToData(data), that triggers a rebuild that may occur before all the horizontal scroll controllers have updated.

File: packages/devtools_app/lib/src/charts/flame_chart.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/charts/flame_chart.dart:456]: done

**Pull request merged**